src/api/java/com/minecolonies/api/colony/IColonyView.java

```diff
@@ -266,6 +270,40 @@ public interface IColonyView extends IColony
     @Nullable
     IMessage handleColonyBuildingViewMessage(BlockPos buildingId, @NotNull FriendlyByteBuf buf);
 
+    /**
+     * Update a ColonyView's fields given a network data ColonyView update packet. This uses a full-replacement - fields do not get updated and are instead overwritten.
+     *
+     * @param type     the field type.
+     * @param position the field position.
+     * @param buf     buffer containing field information.
+     */
+    void handleColonyFieldViewMessage(@NotNull FieldRegistries.FieldEntry type, @NotNull BlockPos position, @NotNull FriendlyByteBuf buf);
```

Comment:
Same thing, this should be in the manager that holds the fields and not fill up the colony view and colony if possible